lib/js/components/DatePickers/DatePicker/DatePicker.vue

```diff
@@ -273,12 +281,16 @@ export default defineComponent({
 			mode: 'single',
 		});
 		watch([() => props.isInteractive, () => props.state], async () => {
-			if (props.isInteractive && props.state === DATE_PICKER_STATES.DEFAULT) {
-				await createDatePicker(
+			if (
+				props.isInteractive &&
+				props.state === DATE_PICKER_STATES.DEFAULT &&
+				!flatpickrInstance.value
+			) {
+				flatpickrInstance.value = (await createDatePicker(
```

Comment:
Is this watch necessary? We don't want to create watch on state change, rather on toggle. Also, there is a difference in assignment. Here you assign an instance to the value property, below you assign to the variable itself

Reply:
We still leave flatpickr when we close it. Maybe we should create/destroy it on toggle and then, probably, we don't need this watcher.